### PR TITLE
Addec command and more

### LINKED ISSUE #31: Improve UI for Emergency Contact fields

### LINKED ISSUE #33: Delete Command for Emergency Contact

### LINKED ISSUE #69: Add command for multiple emergency contacts
Allow user to add additional emergency contacts to users

### PR DESCRIPTION
**Pull request by kellywsq03**
Closes #69, closes #33: Extend the add, edit and delete commands to support multiple emergency contacts.

Closes #31: Gives Emergency Contacts a cool teal color scheme :)

### CODE REVIEW

**Comment by kellywsq03** [comment]
Please don't merge yet, I'll fix the code coverage and comb for bugs

File: src/main/java/seedu/address/commons/core/index/Index.java [hunk 1]
```diff
@@ -66,4 +67,15 @@ public boolean equals(Object other) {
...
+    /**
+     * Represents a constant index used to signify the absence of an emergency contact index
+     * provided in the delete command.
+     * This class extends {@code Index} and provides a predefined index value of {@code Integer.MAX_VALUE}.
+     */
+    public static class EmptyEmergencyContactIndex extends Index {
+        public EmptyEmergencyContactIndex() {
+            super(Integer.MAX_VALUE);
+        }
+    }
```
Comment (Johnwz123) [inline src/main/java/seedu/address/commons/core/index/Index.java:80]: Sorry. Could you explain the use of this nested class?

**Review by Johnwz123 — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/commons/core/index/Index.java (hunk 1 above)
Comment (kellywsq03) [reply src/main/java/seedu/address/commons/core/index/Index.java:80]: This class is used as a placeholder in parsing arguments the absence of an emergency contact index provided in the delete command.

As this introduces a fair bit of complexity, I aim to introduce an ec/ field in the delete command instead, such that this class may be removed.

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java [hunk 2]
```diff
@@ -19,16 +30,66 @@ public class DeleteCommand extends Command {
...
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
-            + ": Deletes the person identified by the index number used in the displayed person list.\n"
-            + "Parameters: INDEX (must be a positive integer)\n"
-            + "Example: " + COMMAND_WORD + " 1";
+            + ": Deletes the person by the index number used in the displayed person list or emergency contact "
+            + "identified by the index number used in the displayed emergency contact list.\n"
+            + "Parameters: INDEX (must be a positive integer) [" + PREFIX_EMERGENCY_CONTACT_TO_EDIT
+            + "EMERGENCY CONTACT INDEX (must be a positive integer)]\n"
+            + "Example: " + COMMAND_WORD + " 1 1";
```
Comment (aldentantan) [inline src/main/java/seedu/address/logic/commands/DeleteCommand.java:37]: Should the example message also include a `PREFIX_EMERGENCY_CONTACT_TO_EDIT` for the second integer?

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java [hunk 3]
```diff
@@ -20,15 +21,36 @@ public class DeleteCommand extends Command {
...
+
+    private CommandResult executeDeleteEmergencyContact(Index emergencyContactIndex,
+                                                        Person personToDelete,
+                                                        Model model) throws CommandException {
+        if (personToDelete.hasOnlyOneEmergencyContact()) {
+            throw new CommandException(Messages.MESSAGE_LAST_EMERGENCY_CONTACT_INDEX);
+        }
+        EmergencyContact deletedEmergencyContact =
+                personToDelete.getAndRemoveEmergencyContact(emergencyContactIndex);
+        model.setPerson(personToDelete, personToDelete);
```
[the remark this answers is not in the document]
Comment (kellywsq03) [reply src/main/java/seedu/address/logic/commands/DeleteCommand.java]: It refreshes the address book to display the updated person!

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java (hunk 2 above)
Comment (kellywsq03) [reply src/main/java/seedu/address/logic/commands/DeleteCommand.java:37]: Will add this in my next pull request. Thanks for the good catch!

**Review by saajidshaik02 — CHANGES_REQUESTED** [review]
Please see below

File: src/main/java/seedu/address/ui/EmergencyContactCard.java
[the remark this answers is not in the document]
Comment (kellywsq03) [reply src/main/java/seedu/address/ui/EmergencyContactCard.java:1]: Great suggestion! Will add it to my next push :)

File: src/main/java/seedu/address/logic/commands/AddEmergencyContactCommand.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+    /**
+     * Creates and returns a {@code Person} with the details of {@code personToAddEmergencyContactTo}
+     * with an added {@code emergencyContactToAdd}.
+     */
+    private static Person createEditedPerson(Person personToEdit, EmergencyContact emergencyContactToAdd) {
+        assert personToEdit != null;
+
+        Set<EmergencyContact> personToEditEmergencyContacts = personToEdit.getEmergencyContacts();
+        Set<EmergencyContact> updatedEmergencyContacts = new LinkedHashSet<>(personToEditEmergencyContacts);
```
[the remark this answers is not in the document]
Comment (kellywsq03) [reply src/main/java/seedu/address/logic/commands/AddEmergencyContactCommand.java:69]: The Delete command operates on the index of the emergency contacts. A HashSet doesn't necessarily preserve the order of the elements, hence resulting in rogue outcomes during deletion. A LinkedHashSet preserves the order and hence provides predictable outcomes! Hope this helps!

**Review by Johnwz123 — APPROVED** [review]

**Review by aldentantan — APPROVED** [review]

**Review by saajidshaik02 — APPROVED** [review]
LGTM!

**Pull request merged**